examples/vision/perceiver_image_classification.py

```diff
@@ -0,0 +1,477 @@
+
+        # Apply global average pooling to generate a [batch_size, projection_dim] repesentation tensor.
+        representation = self.global_average_pooling(latent_array)
+        # Generate logits.
+        logits = self.classification_head(representation)
+        return logits
+
+
+"""
+## Compile, train, and evaluate the mode
+"""
+
+
+def run_experiment(model):
+
+    # Create LAMB optimizer with weight decay.
+    optimizer = tfa.optimizers.LAMB(
+        learning_rate=learning_rate, weight_decay_rate=weight_decay,
+    )
+
+    # Compile the model.
+    model.compile(
+        optimizer=optimizer,
+        loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+        metrics=[
+            keras.metrics.SparseCategoricalAccuracy(name="acc"),
+            keras.metrics.SparseTopKCategoricalAccuracy(5, name="top5-acc"),
+        ],
+    )
+
+    # Create a learning rate scheduler callback.
+    reduce_lr = keras.callbacks.ReduceLROnPlateau(
+        monitor="val_loss", factor=0.2, patience=3
+    )
+
+    # Create an early stopping callback.
+    early_stopping = tf.keras.callbacks.EarlyStopping(
+        monitor="val_loss", patience=15, restore_best_weights=True
+    )
+
+    # Fit the model.
+    history = model.fit(
+        x=x_train,
+        y=y_train,
+        batch_size=batch_size,
+        epochs=num_epochs,
+        validation_split=0.1,
+        callbacks=[early_stopping, reduce_lr],
+    )
+
+    _, accuracy, top_5_accuracy = model.evaluate(x_test, y_test)
+    print(f"Test accuracy: {round(accuracy * 100, 2)}%")
+    print(f"Test top 5 accuracy: {round(top_5_accuracy * 100, 2)}%")
+
+    # Return history to plot learning curves.
+    return history
+
+
+"""
+Note that training the perceiver model with the current settings on a V100 GPUs takes
+around 200 seconds.
+"""
+
+perceiver_classifier = Perceiver(
+    patch_size,
+    num_patches,
+    latent_dim,
+    projection_dim,
+    num_heads,
+    num_transformer_blocks,
+    ffn_units,
+    dropout_rate,
+    num_iterations,
+    classifier_units,
+)
+
+
+history = run_experiment(perceiver_classifier)
+
+"""
+After 50 epochs, the Perceiver model achieves around 53% accuracy and 81% top-5 accuracy on the test data.
+
+As mentioned in the ablations of the [Perceiver](https://arxiv.org/abs/2103.03206) paper,
+you can obtain better results by increasing the latent array size,
+increasing the (projection) dimensions of the latent array and data array elements,
+increasing the number of blocks in the Transformer module, and increasing the number of iterations of applying
+the cross-attention and the latent Transformer modules. You may also try to increase the size the input images
+and use different patch sizes.
+
+The Perceiver benefits from inceasing the model size. However, larger models needs bigger accelerators
+to fit in and train efficiently. This is why in the Perceiver paper they used 32 TPUs to run the experiments.
```

Comment:
"32 TPUs" is a bit unclear, I think they used a TPU pod (with 32 chips)?

Reply:
Fixed